src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -82,19 +85,31 @@
 
         @SuppressWarnings("unchecked") // untypable
         @Override protected Void run() throws Exception {
+            TaskListener l = getContext().get(TaskListener.class);
```

Comment:
:bug: use `@StepContextParameter` so that `StepDescriptor.getRequiredContext` is correctly computed.